packages/aws-cdk-lib/core/lib/context-provider.ts

```diff
@@ -30,16 +30,66 @@ export interface GetContextKeyOptions {
  */
 export interface GetContextValueOptions extends GetContextKeyOptions {
   /**
-   * The value to return if the context value was not found and a missing
-   * context is reported.
+   * The value to return if the lookup has not yet been performed.
+   *
+   * Upon first synthesis, the lookups has not yet been performed. The
+   * `getValue()` operation returns this value instead, so that synthesis can
+   * proceed. After synthesis completes the first time, the actual lookup will
+   * be performed and synthesis will run again with the *real* value.
+   *
+   * Dummy values should preferably have valid shapes so that downstream
+   * consumers of lookup values don't throw validation exceptions if they
+   * encounter a dummy value (or all possible downstream consumers need to
+   * effectively check for the well-known shape of the dummy value); throwing an
+   * exception would error out the synthesis operation and prevent the lookup
+   * and the second, real, synthesis from happening.
+   *
+   * ## Link to ignoreErrorOnMissingContext
+   *
+   * `dummyValue` is also used as the official value to return if the lookup has
+   * failed and `ignoreErrorOnMissingContext` is set.
```

Comment:
Isn't ignoreErrorOnMissingContext not ignoreFailedLookup?

```suggestion
   * ## Link to ignoreFailedLookup
   *
   * `dummyValue` is also used as the official value to return if the lookup has
   * failed and `ignoreFailedLookup` is set.
```

Reply:
Of course, I decided to rename it later. Thanks!